test_shell_error() doesn't catch the right regex on macOS in zsh

Hey, I'm not sure whether macOS or zsh is at fault here, but if you run tox on macOS, the `test_shell_error()` test fails to match the regex:

```
tox -e py311-dj41-sqlite
# snip
.tox/py311-dj41-sqlite/lib/python3.11/site-packages/django_migrations_ci/shell.py:23: MigrateCIShellException

During handling of the above exception, another exception occurred:

    def test_shell_error():
>       with pytest.raises(shell.MigrateCIShellException, match="/bin/sh: 1: oof: not found\n"):
E       AssertionError: Regex pattern did not match.
E        Regex: '/bin/sh: 1: oof: not found\n'
E        Input: '/bin/sh: oof: command not found\n'

tests/test_shell.py:12: AssertionError
```


Here's what I believe will fix it:

```diff
diff --git a/tests/test_shell.py b/tests/test_shell.py
index 56a14c7..d3a0765 100644
--- a/tests/test_shell.py
+++ b/tests/test_shell.py
@@ -9,7 +9,10 @@ def test_shell_ok():


 def test_shell_error():
-    with pytest.raises(shell.MigrateCIShellException, match="/bin/sh: 1: oof: not found\n"):
+    with pytest.raises(
+        shell.MigrateCIShellException,
+        match=r"/bin/sh: (1: )?oof: (command )?not found\n",
+    ):
         shell.exec("oof")
```

If you think this is worth adding in, I'm happy to throw up a quick PR.

--- a/tests/test_shell.py
+++ b/tests/test_shell.py
@@ -9,7 +9,7 @@ def test_shell_ok():
 
 
 def test_shell_error():
-    with pytest.raises(shell.MigrateCIShellException, match="/bin/sh: 1: oof: not found\n"):
+    with pytest.raises(shell.MigrateCIShellException, match=r"oof.+not found"):
         shell.exec("oof")
 
 

